keps/sig-api-machinery/2887-openapi-enum-types/README.md

```diff
@@ -351,6 +351,15 @@ const (
 
 Here, `StorageMedium` can have infinite number of possible values, which disqualify it as an enum type.
 
+The parser of Go compiler and, as a result, `gengo`, does not support type aliases. See https://github.com/kubernetes/gengo/issues/224 for details.
+As of Kubernetes 1.23, we do not have any type aliases in kubernetes/kubernetes repo. Any attempt to add aliases would break the code generation.
+Example: https://github.com/kubernetes/kubernetes/pull/106300
+
+However, as of 1.23, the enum marker is the only marker to be added to a type declaration, and would be the first marker to be affected.
+Until there is a fix to `gengo`, the enum generator has the following limitations:
+- the enum marker must not be added to aliases
+- an aliased enum type or value SHOULD NOT have comments. Otherwise, the comments will be squashed with these of the original with undefined ordering. 
```

Comment:
being able to detect this case is important so people's verify scripts work consistently, right?.

Reply:
Correct. This should be detectable with simple regex matching. I will add a note that in this release cycle either `gengo` gets a fix or I will add a check script to detect this.